system/make.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Linus Unnebäck <linus@folkdatorn.se>
+#
+# This file is part of Ansible
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+# import module snippets
+from ansible.module_utils.basic import *
+
+DOCUMENTATION = '''
+---
+module: make
+short_description: Run targets in a Makefile
+requirements: []
+version_added: "1.9"
+author: Linus Unnebäck (@LinusU) <linus@folkdatorn.se>
+description: Run targets in a Makefile.
+options:
+  target:
+    description: The target to run
+    required: false
+  params:
+    description: Any extra parameters to pass to make
+    required: false
+  chdir:
+    description: cd into this directory before running make
+    required: true
+'''
+
+EXAMPLES = '''
+# Build the default target
+- make: chdir=/home/ubuntu/cool-project
+
+# Run `install` target as root
+- make: chdir=/home/ubuntu/cool-project target=install
+  become: yes
+
+# Pass in extra arguments to build
+- make: chdir=/home/ubuntu/cool-project target=all params='NUM_THREADS=4'
+'''
+
+
+def push_arguments(cmd, args):
+    if args['target'] != None:
+        cmd.append(args['target'])
+    if args['params'] != None:
+        cmd.extend(args['params'].split(' '))
+    return cmd
+
+
+def check_changed(make_path, module, args):
+    cmd = push_arguments([make_path, '--question'], args)
+    rc, _, __ = module.run_command(cmd, check_rc=False, cwd=args['chdir'])
+    return (rc != 0)
+
+
+def run_make(make_path, module, args):
+    cmd = push_arguments([make_path], args)
+    module.run_command(cmd, check_rc=True, cwd=args['chdir'])
+
+
+def main():
+    module = AnsibleModule(
+        supports_check_mode=True,
+        argument_spec=dict(
+            target=dict(required=False, default=None),
+            params=dict(required=False, default=None),
```

Comment:
Would it make sense for params to be a dict type or is it really any free form string?